lib/util/http/github.ts

```diff
@@ -273,12 +278,33 @@ export class GithubHttp extends Http<GithubHttpOptions, GithubHttpOptions> {
     options?: InternalHttpOptions & GithubHttpOptions,
     okToRetry = true
   ): Promise<HttpResponse<T>> {
-    const opts = {
+    const opts: GithubHttpOptions = {
       baseUrl,
       ...options,
       throwHttpErrors: true,
     };
 
+    if (!opts.token) {
```

Comment:
In what scenarios would `token` already be passed here?

Reply:
from `initPlatform`

https://github.com/renovatebot/renovate/blob/f1143dcbf7e1f8d5e95a1832c7dcb6bea3c1a3af/lib/modules/platform/github/index.ts#L143-L166